portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,7 +29,7 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> msgs = new ArrayList<String>();
+    List msgs = new ArrayList<>();
```

Comment:
Sorry, I wasn't completely clear.
You want to have the type be List<String>.
The line should be
  List<String> msgs = new ArrayList<>();
where the <> infers the type from the List<String>.

Reply:
Sorry, I'm a bit confused because I think I've made that change. For me, I have `List msgs = new ArrayList<>()` at line 32. Maybe it didn't update with my latest push?

Reply:
Sorry, HTML formatting got in the way of the actual comment.
You still want the generic type parameter in the List type. So the line should be:
  List&lt;String&gt; msgs = new ArrayList&lt;&gt;();

Reply:
Oh! I see, thanks!